### PR TITLE
feat: agent command parity — cleanup dropped agents, fix plugin catalog, add check-agents CLI

### PR DESCRIPTION
**Pull request by harshanandak**
## Summary
- **Remove all debris from 4 dropped agents** (Antigravity, Windsurf, Aider, Continue): deleted files, code paths, config entries, doc references, gitignore entries
- **Fix plugin catalog** for 6 agents: correct `capabilities.commands` and `directories` flags that were stale after PR #57 added sync-generated command files
- **Add `forge check-agents` CLI** (`node scripts/check-agents.js`): validates all agent command files are in sync and plugin catalog matches reality
- Net result: **-448 lines** (1220 added, 1668 removed) — repo is cleaner

## Design Doc
See: [docs/plans/2026-03-15-agent-command-parity-v2-design.md](docs/plans/2026-03-15-agent-command-parity-v2-design.md)

## Decisions Log
See: [docs/plans/2026-03-15-agent-command-parity-v2-decisions.md](docs/plans/2026-03-15-agent-command-parity-v2-decisions.md) — 0 decision gates fired (plan quality: Excellent)

## Beads Issue
Closes: forge-2w3

## Key Decisions

1. **Delete vs edit rule**: Files fundamentally about dropped agents → delete entirely. Files with minor mentions → edit to remove references. Historical docs (CHANGELOG, design docs) left untouched.
2. **Plugin catalog**: Set `commands: true` for all 8 supported agents since PR #57's sync script generates command files for all of them. Set `hooks: true` for Claude Code.
3. **check-agents CLI reuses sync infrastructure**: Delegates to `syncCommands({ check: true })` rather than reimplementing file validation. Adds plugin catalog cross-check on top.
4. **Gitignore cleanup**: Removed entries for dropped agent dirs (`.agents/`, `.agent/`, `.aider/skills/`, `.continue/skills/`, `.windsurf/skills/`) — cleaner repo, no false safety net for agents that aren't supported.
5. **docs/README-v1.3.md deleted**: Frozen v1.3 snapshot with Antigravity/Windsurf/Continue references throughout — misleading for any agent reading it.

## TDD Test Coverage
- New tests: 7 test files (dropped-agent-files, dropped-agent-code, dropped-agent-cli, dropped-agent-docs, dropped-agent-config, plugin-catalog, check-agents)
- Total suite: **1437 tests, 1406 pass, 31 skip, 0 fail**
- check-agents: 11 tests covering happy path, sync drift, plugin validation, edge cases
- Cleanup tests: grep-based assertions ensuring zero dropped-agent references in active code

## Security Review
- OWASP Top 10: N/A — purely file cleanup and metadata correction, no user input processing, no auth, no network calls
- No new dependencies added

## Test Plan
- [x] Lint: 0 errors, 0 warnings (ESLint --max-warnings 0)
- [x] Tests: 1437 tests, 0 fail
- [x] `node scripts/sync-commands.js --check` → "All agent command files are in sync."
- [x] `node scripts/check-agents.js` → "All agent checks passed."
- [x] Blast-radius grep: zero dropped-agent references in active code/docs

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details><summary><h3>Greptile Summary</h3></summary>

This PR completes the "agent command parity" cleanup by removing all code, config, docs, and files for four dropped agents (Antigravity, Windsurf, Aider, Continue), correcting stale `capabilities.commands` flags in six plugin catalog entries, and introducing a new `forge check-agents` CLI (`scripts/check-agents.js`) that validates sync state and plugin catalog consistency in CI.

All feedback from the previous review round has been addressed: the Codex exemption is now ID-scoped (`plugin.id === 'codex'`), `syncCommands()` is wrapped in a try/catch, an early `return` was added when `syncResult.empty` is true, and the `bin/forge-cmd.js` stage count and the `bin/forge.js` Continue-regex were both tightened.

Key observations:

- **`check-agents.js` logic is sound**: the two-phase check (sync state → plugin catalog) correctly delegates per-file diffing to `syncCommands({ check: true })` and only adds a directory-level sanity check on top. All SyncEntry fields accessed (`agent`, `dir`, `filename`, `filePath`) are present in the typedef and confirmed populated at the push site in `sync-commands.js`.
- **Plugin directory mappings are consistent**: cross-checked each plugin's `directories` key against the corresponding `AGENT_ADAPTERS` entry in `scripts/sync-commands.js` — `cline → .clinerules/workflows/`, `kilo-code → .kilocode/workflows/`, `github-copilot → .github/prompts/`, `roo-code → .roo/commands/`, `cursor → .cursor/commands/`, `opencode → .opencode/commands/`, `codex → .codex/skills/<name>/` all match their plugin JSON entries.
- **Minor style issues only**: `test/cleanup/dropped-agent-cli.test.js` contains four individual tests for Continue patterns and then a fifth combined test that repeats all four — creating redundant assertions. `scripts/check-agents.js` passes an explicit `dryRun: false` that is logically unnecessary since `check: true` causes `syncCommands` to enter the check branch regardless.
</details>

<details><summary><h3>Confidence Score: 4/5</h3></summary>

- Safe to merge — changes are pure cleanup and metadata correction with no runtime-critical logic gaps.
- All previous review issues have been resolved, plugin directory mappings are consistent with their adapters, the new check-agents CLI is well-tested (11 tests), and 1437 tests pass with 0 failures. The only remaining items are two minor style observations (redundant test assertions and a redundant `dryRun: false` argument) that have no correctness impact.
- No files require special attention. The new scripts/check-agents.js is the most complex new file and it reads cleanly.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| scripts/check-agents.js | New CLI validator that delegates sync checking to syncCommands and then validates plugin catalog. All previous review feedback (try/catch, early-return on empty, Codex ID-specific exemption) has been addressed. Minor: dryRun: false is redundant but harmless since check: true always enters the check path before dryRun would matter. |
| packages/skills/src/lib/agents.js | Cleanly removes aider, antigravity, continue, windsurf from AGENT_DEFINITIONS; remaining agents unchanged. |
| lib/agents/cline.plugin.json | commands: true added with directories.workflows: ".clinerules/workflows" — matches the cline adapter's baseDir in sync-commands.js. Correct. |
| lib/agents/codex.plugin.json | commands: true added, skills: false (previously incorrect), directories.skills added to match codex adapter's .codex/skills path; ID-specific exemption in check-agents.js handles the non-standard directory key. Consistent with prior review decisions. |
| bin/forge.js | Removes continueFormat property, setupContinueMcpConfig helper, and displayMcpStatus Continue entry cleanly. copyAgentCommands guard updated to drop continueFormat check. |
| test/scripts/check-agents.test.js | Comprehensive tests for checkAgents: happy path, sync drift, stale files, missing manifest, plugin catalog edge cases. Temp dir teardown is done correctly in finally blocks. The "returns correct shape with current repo root" test is shape-only which is fine for shape testing but doesn't verify correctness of the actual repo state in CI. |
| test/cleanup/dropped-agent-cli.test.js | Tests for Continue/Aider removal from bin/forge.js and bin/forge-cmd.js. Contains redundant assertions (individual tests for continueFormat, continueConfig, etc. AND a combined test that repeats the same four pattern checks). |
| packages/skills/src/commands/sync.js | Removes updateAiderConfig helper and special Aider config-file handling. Supported agents list in the "no agents detected" message updated to reflect 7 remaining agents. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A([node scripts/check-agents.js]) --> B[checkAgents repoRoot]

    B --> C{try syncCommands\ncheck:true}
    C -->|throws| D[errors.push\nSync check failed\nreturn early]
    C -->|ok| E{syncResult.empty?}
    E -->|yes| F[errors.push\nNo command files\nreturn early]
    E -->|no| G{manifestMissing?}
    G -->|yes| H[warnings.push\nmanifest not found]
    G -->|no| I[continue]
    H --> I
    I --> J{outOfSync\nentries?}
    J -->|yes| K[errors.push\nOut of sync per entry\nmissing or modified]
    J -->|no| L[continue]
    K --> L
    L --> M{staleFiles?}
    M -->|yes| N[errors.push\nStale file per path]
    M -->|no| O[Plugin catalog check]
    N --> O

    O --> P{lib/agents/\nexists?}
    P -->|no| Q[warnings.push\nno plugin dir]
    P -->|yes| R[read *.plugin.json]
    R --> S{for each plugin\ncommands:true?}
    S -->|no| T[skip]
    S -->|yes| U{commandDir\nin directories?}
    U -->|no, codex+skills| V[skip Codex]
    U -->|no, other| W[warnings.push\nno dir configured]
    U -->|yes| X{dir exists\non disk?}
    X -->|no| Y[errors.push\ndir missing]
    X -->|yes| Z{dir has\nfiles?}
    Z -->|no| AA[errors.push\ndir empty]
    Z -->|yes| AB[pass]

    T --> AB
    V --> AB
    W --> AB
    Y --> AB
    AA --> AB
    AB --> AC{errors.length?}
    AC -->|0| AD([exit 0\nAll checks passed])
    AC -->|>0| AE([exit 1\nN errors found])
```
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: test/cleanup/dropped-agent-cli.test.js
Line: 44-50

Comment:
**Redundant combined test repeats individual assertions**

Lines 36–43 already have four individual `test(...)` cases — one for each of `continueFormat`, `continueConfig`, `generateContinueConfig`, and `.continue/`. This fifth test then re-checks all four patterns again as a single block. If any pattern regresses, two tests will fail simultaneously, which can be confusing.

Consider removing the combined test entirely and keeping only the individual, named tests — they already provide the same coverage with better failure messages:

```suggestion
```

(Remove lines 44–50; the four individual tests above are sufficient.)

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: scripts/check-agents.js
Line: 40

Comment:
**Redundant `dryRun: false` argument**

`syncCommands` checks `if (dryRun)` before `if (check)`, so passing `dryRun: false` is logically equivalent to omitting it (any falsy value skips the dry-run path). The explicit `false` is harmless today, but if the function signature ever gains a default value for `dryRun`, this could shadow it unexpectedly. Omitting the argument keeps the call site cleaner and self-documenting:

```suggestion
    syncResult = syncCommands({ check: true, repoRoot });
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Last reviewed commit: be3da11</sub>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,414 +1 @@
...
-**Ready to start?**
-
-```bash
-bun install forge-workflow
-bunx forge setup
-/status
-```
-
-Then open [QUICKSTART.md](QUICKSTART.md) and ship your first feature! 🚀
+# Test Repository
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply README.md]: This is a false positive — README.md was not modified in this PR.

`git diff master -- README.md` shows zero changes. The placeholder text (`# Test Repository`) exists only inside `test-env/validation/git-state-checker.test.js` which writes it to a temporary directory during test fixture setup — it never touches the actual repo README.

Evidence: The PR's file list (44 files changed) does not include README.md.

File: lib/agents/codex.plugin.json [hunk 2]
```diff
@@ -5,11 +5,13 @@
...
-    "commands": false,
-    "skills": false,
+    "commands": true,
+    "skills": true,
     "hooks": false
   },
-  "directories": {},
+  "directories": {
+    "skills": ".codex/skills"
+  },
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/agents/codex.plugin.json:14]: This is by design — Codex CLI does not support project-level slash commands (`/plan`, `/dev`, etc.). It uses Skills instead, where each skill lives in `.codex/skills/<name>/SKILL.md`.

The `directories.skills` mapping to `.codex/skills` is correct and matches the actual file structure on disk (11 skill directories exist). The check-agents validator's Codex exemption (line 114) exists because Codex's command mechanism IS skills — adding a `commands` key would be misleading since there's no `.codex/commands/` directory.

See: docs/plans/2026-03-15-agent-command-parity-v2-design.md and the adapter config in scripts/sync-commands.js (line 242).

File: scripts/check-agents.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+          continue;
+        }
+        warnings.push(
+          `Plugin "${plugin.name}" (${plugin.id}) has commands: true but no command directory configured.`
+        );
+        continue;
+      }
+
+      const absCmdDir = path.join(repoRoot, commandDir);
+
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply scripts/check-agents.js:122]: ✅ Fixed: Tightened the Codex exemption to only match plugin.id === 'codex'

Changed: `if (dirs.skills)` → `if (plugin.id === 'codex' && dirs.skills)`
Reason: Prevents future misconfigured plugins from silently passing validation
Commit: ea2a3f2

File: bin/forge-cmd.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ const VALID_COMMANDS = [
 
 const COMMAND_DESCRIPTIONS = {
 	status: 'Detect current workflow stage (1-9)',
-	plan: 'Create branch + Beads + OpenSpec proposal',
+	plan: 'Create branch + Beads + design doc',
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply bin/forge-cmd.js:34]: ✅ Fixed: Updated status command description from (1-9) to (1-7)

Changed: `'Detect current workflow stage (1-9)'` → `'Detect current workflow stage (1-7)'`
Reason: Consistent with the 7-stage workflow described in package.json and AGENTS.md
Commit: ea2a3f2

File: scripts/check-agents.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * Validate all agent configs are complete and in sync.
+ *
+ * @param {string} repoRoot - Absolute path to the repository root
+ * @returns {{ errors: string[], warnings: string[] }}
+ */
+function checkAgents(repoRoot) {
+  /** @type {string[]} */
+  const errors = [];
+  /** @type {string[]} */
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply scripts/check-agents.js:34]: ✅ Fixed: Wrapped syncCommands() in try/catch — filesystem errors now produce clean output in errors[] instead of raw stack traces.
Commit: 65a0058

File: lib/agents/codex.plugin.json [hunk 6]
```diff
@@ -5,11 +5,13 @@
   "description": "OpenAI's terminal coding agent",
   "homepage": "https://github.com/openai/codex",
   "capabilities": {
-    "commands": false,
-    "skills": false,
+    "commands": true,
+    "skills": true,
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/agents/codex.plugin.json]: ✅ Fixed: Set `capabilities.skills: false` for Codex.

Codex's `.codex/skills/` files are generated by the command sync path (`scripts/sync-commands.js`), not the skills-sync system (`packages/skills/`). The `skills: true` flag was misleading.
Commit: 65d73e3

File: test/cleanup/dropped-agent-cli.test.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  test('bin/forge.js must not reference Continue AI agent setup', () => {
+    // Match "Continue" as a standalone word (agent name), not the JS keyword "continue;"
+    // Look for patterns like "Continue config", "for Continue", "Setup ... Continue"
+    expect(forgeJs).not.toMatch(/\bContinue\b(?!\s*\()/);
+  });
+
+  // --- bin/forge-cmd.js ---
+
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply test/cleanup/dropped-agent-cli.test.js]: ✅ Fixed: Replaced broad `/\bContinue\b/` regex with specific pattern matches: `.continue/`, `continueFormat`, `continueConfig`, `generateContinue`.

This avoids false failures on natural English like "Continue to the next step" while still catching all agent-specific references.
Commit: 65d73e3

File: scripts/check-agents.js [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    errors.push('No command files found in .claude/commands/ — cannot verify sync.');
+  } else {
+    if (syncResult.manifestMissing) {
+      warnings.push(
+        'Sync manifest (.forge/sync-manifest.json) not found — stale file detection skipped. ' +
+        'Run "node scripts/sync-commands.js" to generate it.'
+      );
+    }
+
+    if (syncResult.outOfSync && syncResult.outOfSync.length > 0) {
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply scripts/check-agents.js:57]: ✅ Fixed: Added early `return { errors, warnings }` when `syncResult.empty` is true.

Now when no command files exist in `.claude/commands/`, the user sees only the root cause error instead of a cascade of plugin directory errors.
Commit: 65d73e3
